crypto/crypto/src/hash.rs

```diff
@@ -356,6 +356,30 @@ impl<'a> std::iter::DoubleEndedIterator for HashValueBitIterator<'a> {
 
 impl<'a> std::iter::ExactSizeIterator for HashValueBitIterator<'a> {}
 
+impl std::iter::FromIterator<bool> for HashValue {
+    fn from_iter<I: IntoIterator<Item = bool>>(iter: I) -> Self {
+        let mut count = 0;
+        let mut buf = vec![];
+        for bit in iter {
+            let index = count % 8;
+            if index == 0 {
+                buf.push(0);
+            }
+            if bit {
+                *buf.last_mut().expect("The last byte must exist.") |= 1 << (7 - index);
+            }
+            count += 1;
+        }
+        HashValue::from_slice(&buf).unwrap_or_else(|_| {
+            panic!(
```

Comment:
A little worrying, not sure if it opens us up to potential attacks. Would rather be verbose by having 

    HashValue::from_bit_iter(bits.into_iter())?;